packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -646,9 +658,9 @@ class _CupertinoBackGestureController<T> {
       controller.removeStatusListener(_handleStatusChanged);
     }
     _animating = false;
+    onEnded();
```

Comment:
Not sure why we now want to do this first. Previously it seemed like effectively an else clause. Why would we want to unconditionally dispose the route before removing it?

Reply:
I thought I wrote it somewhere in the PR but didn't. This is a bug on everything downstream from didStopUserGesture such as NavigatorObserver.didStopUserGesture or NavigatorState.userGestureInProgress. It never worked because the route would be disposed before the back controller would be disposed. We were just lucky that those APIs weren't really used.

Added new tests.